### PR TITLE
Add Xpubs for all BIP44 xprivs

### PR DESCRIPTION
**Pull request by sedited**
To allow users to check their balances through a watch only wallet allow
the exporting of extended public keys.
@benma

### CODE REVIEW

File: js/backup_in.js [hunk 1]
```diff
@@ -362,12 +390,18 @@ function backup_generate()
         var bip32_keychain = new BIP32(xpriv);
         var electrum_key_btc_leg = bip32_keychain.derive("m/44'/0'/0'");
         electrum_key_btc_leg = base58check.encode(new Buffer(electrum_key_btc_leg.extended_private_key, 'hex'));
+        var electrum_key_btc_leg_pub = bip32_keychain.derive("m/44'/0'/0'");
```
Comment (benma) [inline js/backup_in.js:393]: why not drop this line and use `electrum_key_btc_leg.extended_public_key` in the following line?

**Review by benma — CHANGES_REQUESTED** [review]

File: js/backup_in.js (hunk 1 above)
Comment (sedited) [reply js/backup_in.js:393]: I tried that when writing this patch, but turns out once it is encoded in hex in 392 it can't be reused.

**Comment by NickeZ** [comment]
Does this include QR codes as well?

**Comment by douglasbakkum** [comment]
@TheCharlatan lgtm

@NickeZ what do you mean? display a QR code for each xpriv or xpub derivation? that would be convenient but better for a different PR and not sure it would be worth the effort (i.e. lost opportunity cost on other items) at this time.

**Comment by sedited** [comment]
No qr code included, but the user could enter the xpub into quite about any qr code generator and at least the watch only wallets that I tested are then able to correctly import it.

**Comment by NickeZ** [comment]
OK, cool

File: js/backup_in.js (hunk 1 above)
Comment (benma) [reply js/backup_in.js:393]: that's fucked up

**Review by benma — APPROVED** [review]
tested ack

**Pull request merged**